pandas/io/parsers.py

```diff
@@ -1770,9 +1771,22 @@ def _handle_usecols(self, columns, usecols_key):
                     raise ValueError("If using multiple headers, usecols must "
                                      "be integers.")
                 col_indices = []
+                # if duplicates exist, save the original usecols_key. GH11823
```

Comment:
use ``Index(...)`` and then use the proper index operations here instead hand rolling this.